osu.Game/Screens/Select/SongSelect.cs

```diff
@@ -203,8 +204,17 @@ private void editRequested(BeatmapInfo beatmap)
             Push(new Editor());
         }
 
-        private void onBeatmapRestored(BeatmapInfo b) => Schedule(() => carousel.UpdateBeatmap(b));
-        private void onBeatmapHidden(BeatmapInfo b) => Schedule(() => carousel.UpdateBeatmap(b));
+        private void onBeatmapRestored(BeatmapInfo beatmap)
+        {
+            var beatmapSet = beatmaps.QueryBeatmapSet(s => s.ID == beatmap.BeatmapSetInfoID);

```

Comment:
why is this not also scheduled? this means it is going to user the database context of whichever thread this was called on (which is likely incorrect).

Reply:
Mh. There shouldn't be anything against putting the query in the schedule. I'll check moving it.